src/lib/badge/badge.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+ import {Component, Input, ViewEncapsulation, Renderer2, ElementRef} from '@angular/core';
+ import {coerceBooleanProperty} from '@angular/cdk/coercion';
+ import {CanColor, mixinColor} from '@angular/material/core';
+
+ /** @docs-private */
+export class MatBadgeBase {
+  constructor(public _renderer: Renderer2, public _elementRef: ElementRef) {}
+}
+export const _MatBadgeMixinBase = mixinColor(MatBadgeBase);
+
+@Component({
+  selector: 'mat-badge',
+  templateUrl: './badge.html',
+  styleUrls: ['./badge.css'],
+  inputs: ['color'],
+  host: {
+    'class': 'mat-badge',
+    '[class.mat-badge-overlap]': 'overlap',
+    '[class.mat-badge-above]': '_isAbove',
+    '[class.mat-badge-below]': '!_isAbove',
+    '[class.mat-badge-before]': '_isBefore',
+    '[class.mat-badge-after]': '!_isBefore',
+  },
+  encapsulation: ViewEncapsulation.None,
+  preserveWhitespaces: false,
+})
+export class MatBadge extends _MatBadgeMixinBase implements CanColor {
+
+  /** Whether the badge should overlap its contents or not */
+  @Input()
+  set overlap(val: boolean) {
+    this._overlap = coerceBooleanProperty(val);
+  }
+  get overlap(): boolean {
+    return this._overlap;
+  }
+  private _overlap: boolean = true;
+
+  /** Content of the badget */
+  @Input() content: string = '';
+
+  /** Direction the badge should reside; 'before|after above|below' */
+  @Input() direction: string = 'above after';
```

Comment:
The default value here is inconsistent with how the `_isBefore` getter works. If I don't pass a direction, it will end up as `above after`. If I use `"banana banana"`, it will be `below after`.

As a nit, the overview guide implies there is an order to which the directions should be in, but that order is reversed in the jsdoc comment.

Reply:
I made the getters more specific to prevent the monkeys ;P

Thanks for the heads up. Fixed that too.